### ISSUE TITLE
go/types, types2: errors for absolute import path don't match (old) compiler, and are inconsistent with the compiler

### ISSUE THREAD

**Issue by griesemer** [golang/go#43109]
For
```Go
package main
import "/foo"
```
the compiler reports:
```
main.go:2:8: import path cannot be absolute path
```
Both `go/types` and `types2` report:
```
main.go:2:8: could not import /foo (can't find import: "/foo")
main.go:2:8: "/foo" imported and not used
```

Generally, should review all import path errors.

cc: @findleyr

**Comment by odeke-em** [comment]
Thank you for filing this issue @griesemer and Happy New Year to you and @findleyr!

I am currently at 90860e0c3110ac5898dfe8e0e0fafd0aea8d979a and it seems fixed; the result of this compilation using types2 seems to produce the desired result like this
```shell
go version devel go1.18-90860e0c31 Sat Jan 8 00:24:25 2022 +0000 darwin/amd64
package main
import "/foo"
main.go:2:8: could not import /foo (import path cannot be absolute path)
main.go:2:8: imported and not used: "/foo"
```

Shall we perhaps close this issue?

**Comment by griesemer** [comment]
We're getting the correct `could not import` error, but we shouldn't get an `imported and not used` error (after all it couldn't be imported in the first place). This still needs to be addressed but it's not crucial for 1.18. Moving to 1.19.
